meson.build

```diff
@@ -0,0 +1,1629 @@
+             dependency('xrandr', version: '>= 1.2.0', required: get_option('x11'))],
+}
+foreach dep: x11['deps']
+    if not dep.found()
+        x11 += {'use': false}
+        break
+    endif
+    if dep == x11['deps'][4]
+       x11 += {'use': true}
+    endif
+endforeach
+if x11['use']
+    dependencies += x11['deps']
+    sources += ['video/out/vo_x11.c',
+                'video/out/x11_common.c']
+endif
+
+xv = dependency('xv', required: get_option('xv'))
+if xv.found()
+    dependencies += xv
+    sources += 'video/out/vo_xv.c'
+endif
+
+
+# Opengl dependencies
+gl = {
+    'name': 'gl',
+    'use': false,
+}
+if not get_option('gl').disabled()
+    sources += ['video/out/opengl/common.c',
+                'video/out/opengl/context.c',
+                'video/out/opengl/formats.c',
+                'video/out/opengl/libmpv_gl.c',
+                'video/out/opengl/ra_gl.c',
+                'video/out/opengl/utils.c']
+    plain_gl = {
+        'name': 'plain-gl',
+        'use': not get_option('plain-gl').disabled(),
+    }
+    if plain_gl['use']
+        gl += {'use': true}
+    endif
+
+    egl = {
+        'name': 'egl',
+        'deps': dependency('egl', version: '> 1.4.0', required: get_option('egl')),
+        'use' : false
+    }
+    if egl['deps'].found()
+        dependencies += egl['deps']
+        egl += {'use': true}
+    endif
+
+    egl_drm = {
+        'name': 'egl-drm',
+        'use': drm['use'] and gbm['use'] and egl['use'] and not get_option('egl-drm').disabled(),
+    }
+    if egl_drm['use']
+        gl += {'use': true}
+        sources += 'video/out/opengl/context_drm_egl.c'
+    endif
+
+    egl_wayland = {
+        'name': 'egl-wayland',
+        'deps': dependency('wayland-egl', version: '>= 9.0.0', required: get_option('egl-wayland')),
+    }
+    egl_wayland += {'use': wayland['use'] and egl['use'] and egl_wayland['deps'].found()}
+    if egl_wayland['use']
+        dependencies += egl_wayland['deps']
+        gl += {'use': true}
+        sources += 'video/out/opengl/context_wayland.c'
+    endif
+
+    gl_x11 = {
+        'name': 'gl_x11',
+        'deps': cc.find_library('GL', required: get_option('gl-x11')),
+        'use': false,
+    }
+    if x11['use']
+        if gl_x11['deps'].found()
+            gl_x11 += {'use': true}
+        endif
+    endif
+    if gl_x11['use']
+        dependencies += gl_x11['deps']
+        gl += {'use': true}
+        sources += 'video/out/opengl/context_glx.c'
+    endif
+
+    egl_x11 = {
+        'name': 'egl-x11',
+        'use': x11['use'] and egl['use'] and not get_option('egl-x11').disabled(),
+    }
+    if egl_x11['use']
+        gl += {'use': true}
+        sources += 'video/out/opengl/context_x11egl.c'
+    endif
+
+    if egl_x11['use'] or gl_x11['use']
+        sources += 'video/out/opengl/oml_sync.c'
+    endif
+
+    gl_win32 = {
+        'name': 'gl-win32',
+        'deps': cc.find_library('opengl32', required: get_option('gl-win32')),
```

Comment:
... and handles this for you automatically on Windows.

(`gl.type_name()` will return "system" if it didn't find a pkg-config version, but instead found opengl32 on Windows or the OpenGL framework on Darwin.)